server/game/GameActions/ConditionalGameAction.ts

```diff
@@ -0,0 +1,66 @@
+import { GameAction, GameActionProperties } from './GameAction';
+import AbilityContext = require('../AbilityContext');
+
+export interface ConditionalActionProperties extends GameActionProperties {
+    gameAction: GameAction;
+    condition: boolean | ((context: AbilityContext) => boolean);
+    else?: GameAction;
+}
+
+export class ConditionalGameAction extends GameAction {
+    effect = 'do a conditional action';
```

Comment:
I think we should probably give this an effect message which tells the player what it's going to do